### ISSUE TITLE
[XamlC] Collection was modified; enumeration operation may not execute.

### ISSUE THREAD

**Issue by drasticactions** [dotnet/maui#17563]
### Description

If you have `<?xaml-comp compile="true" ?>` enabled in your ResourceDictionary, it will fail to compile with a enumeration error.

https://github.com/davidortinau/ControlGallery/blob/main/src/ControlGallery/Resources/Styles/DefaultStyles.xaml#L1-L6

### Steps to Reproduce

Add a resourcedictonary file and set XamlC to compile.

### Link to public reproduction project repository

https://github.com/davidortinau/ControlGallery

### Version with bug

8.0.0-rc.1.9171

### Is this a regression from previous behavior?

Yes, this used to work in .NET MAUI

### Last version that worked well

Unknown/Other

### Affected platforms

iOS, Android, Windows, macOS

### Did you find any workaround?

Disable XamlC

### Relevant log output

```shell
Resources\Styles\Text.xaml : XamlC error : Collection was modified; enumeration operation may not execute. [X:\Code\Wor
k\ControlGallery\src\ControlGallery\ControlGallery.csproj::TargetFramework=net8.0-maccatalyst]
Resources\Styles\Text.xaml : XamlC error : Collection was modified; enumeration operation may not execute. [X:\Code\Wor
k\ControlGallery\src\ControlGallery\ControlGallery.csproj::TargetFramework=net8.0-ios]
Resources\Styles\Text.xaml : XamlC error : Collection was modified; enumeration operation may not execute. [X:\Code\Wor
k\ControlGallery\src\ControlGallery\ControlGallery.csproj::TargetFramework=net8.0-android]
```

**Comment by jfversluis** [comment]
Same/related to #17461 ? Where something similar is also mentioned

**Comment by StephaneDelcroix** [comment]
> Same/related to #17461 ? Where something similar is also mentioned

no, but I think I already fixed that

**Comment by StephaneDelcroix** [comment]
dupe of #17333, fixed by #17364
